packages/dd-trace/src/appsec/iast/index.js

```diff
@@ -40,6 +42,11 @@ function onIncomingHttpRequestStart (data) {
           createTransaction(rootSpan.context().toSpanId(), iastContext)
           overheadController.initializeRequestContext(iastContext)
         }
+        if (rootSpan.addTags) {
```

Comment:
Can't we do the same in `onIncomingHttpRequestEnd`?

Reply:
It is possible but much uglier because if iast didn't adquire request on httpRequestStart we need to obtain the `topContext` to access the `span`.
Something like that:

```
if (iastContext && iastContext.rootSpan) {
  sendVulnerabilities(iastContext, iastContext.rootSpan)
  removeTransaction(iastContext)
  addIastEnabledTag(iastContext.rootSpan, true)
} else {
  const topContext = web.getContext(data.req)
  if (topContext) {
    addIastEnabledTag(topContext.span, false)
  }
}
```